backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -13,17 +14,32 @@ import javax.transaction.Transactional
 @Service
 class SpecificationDatabaseService constructor(
     private val specificationRepository: SpecificationRepository,
-    private val entityManager: EntityManager
+    private val entityManager: EntityManager,
+    private val specificationEntityMapper: SpecificationEntityMapper
 ) : SpecificationPersistenceService {
 
     override fun save(specification: Specification) {
-        specificationRepository.save(SpecificationEntity.fromDomain(specification))
+        val specificationEntity = specificationEntityMapper.fromDomain(specification)
+
+        if (specificationRepository.existsByTitle(specificationEntity.title)) {
+            val existingEntity = specificationRepository.findByTitle(specificationEntity.title)
+
+            val versions = existingEntity.versions + specificationEntity.versions
+            val newEntity = SpecificationEntity(existingEntity.id, existingEntity.title, existingEntity.description, versions, existingEntity.remoteAddress)
+            newEntity.versions.map { versionEntity -> versionEntity.specification = newEntity }
+
+            specificationRepository.save(newEntity)
```

Comment:
Maybe the whole block should be a new method?

Reply:
What do you mean exactly?

Reply:
I think that the method does too much and would rather extract the stuff from the `if`-block into a new method.